webui/webui/components/chat.py

```diff
@@ -82,7 +82,8 @@ def action_bar() -> rx.Component:
                     ),
                     is_disabled=State.processing,
                 ),
-                on_submit=State.process_question,
+                on_submit=[State.process_question,rx.set_value("question", "")],
+                # on_submit=[State.baidu_process_question,rx.set_value("question", "")],
```

Comment:
instead of doing this, we could make the apis(that of openai and badu) pluggable, in other words being able to switch between what api to use. I would typically add a config variable in `rxconfig.py` and then set the default to openai (just so the openai api is the default and can be set to badu if need be). 

In the state you can create a new handler `process_openai_question` and move the logic of openapi there, then in the `State.process_question` you choose what handler to trigger based on the config variable set. That seems like a cleaner approach.

```python
from rxconfig import config
...
class State(rx.State):
   ...
   
   async def process_question(...):
        if config.api_type == "openai":
           # trigger State.process_openai_question handler
        else:
            # trigger State.process_badu_question handler